packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -567,6 +574,12 @@ export class AnchorSet implements AnchorLocator {
 			node.parentIndex += destination.parentIndex - coupleInfo.startParentIndex;
 			node.parentPath = destinationPath;
 			node.parentField = destination.parentField;
+			// If the destination is a DetachedUpPath, propagate its detachedNodeId
+			// TODO does the id need to be incremented? it doesn't seem like it so are there ever multiple nodes that get passed here and if so, how do we handle this?
```

Comment:
They need to be offset (not incremented, because the `coupleInfo.nodes` here can be sparse compared to the sequence of nodes being detached in the forest).
It's the same reason why `node.parentIndex` can't just be incremented for each `node`.

You're correct that we don't currently call this function with more than 1 node in `coupleInfo.nodes`. This is a consequence of the fact that we atomize all changes in the delta visit. This function's implementation shouldn't assume that.
It would be good to add a test to the AnchorSet suite that detaches nodes [a, (b), c, d, (e)] where (x) represents a node that the AnchorSet has an anchor for. This way we can check this logic works properly.

Reply:
Note that you also need to clear out the `detachedNodeId` of all those nodes when `destination` is not a `DetachedUpPath`.
A good test for that would be to do the reverse of the test I described above.